pkg/prowgen/prowgen.go

```diff
@@ -146,7 +126,68 @@ func GenerateJobs(configSpec *cioperatorapi.ReleaseBuildConfiguration, info *Pro
 		PresubmitsStatic:  presubmits,
 		PostsubmitsStatic: postsubmits,
 		Periodics:         periodics,
+	}, nil
+}
+
+func generateFromImages(imageTargets sets.String,
+	configSpec *cioperatorapi.ReleaseBuildConfiguration,
+	info *ProwgenInfo,
+	presubmits map[string][]prowconfig.Presubmit,
+	postsubmits map[string][]prowconfig.Postsubmit,
+	newJobBaseBuilder func() *prowJobBaseBuilder) error {
+	if len(imageTargets) == 0 {
+		return nil
+	}
+
+	orgrepo := fmt.Sprintf("%s/%s", info.Org, info.Repo)
+	// Identify which jobs need to have a release payload explicitly requested
+	var presubmitTargets = imageTargets.List()
+	if api.PromotesOfficialImages(configSpec, api.WithOKD) {
+		presubmitTargets = append(presubmitTargets, "[release:latest]")
+	}
+
+	jobBaseGen := newJobBaseBuilder().TestName("images")
+	jobBaseGen.PodSpec.Add(Targets(presubmitTargets...))
+	presubmits[orgrepo] = append(presubmits[orgrepo], *generatePresubmitForTest(jobBaseGen, "images", info))
+
+	generatePostsubmitsJobsForArch := func(arch api.Architecture) error {
```

Comment:
Do we need to do this separately? We already generate the postsubmit. We could pass the additional arch config and generate a list of postsubmits instead.